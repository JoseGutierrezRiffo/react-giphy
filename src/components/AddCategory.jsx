import React, { Fragment, useState } from "react";
import PropTypes from "prop-types";

const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handlerSubmit = e => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {
      setCategories(category => [ inputValue, ...category ]);
      setInputValue("");
    }
  };

  return (
    <Fragment>
      <form onSubmit={handlerSubmit}>
        <input
          type="text"
          value={inputValue}
          onChange={e => setInputValue(e.target.value)}
        />
      </form>
    </Fragment>
  );
};

export default AddCategory;

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
};
