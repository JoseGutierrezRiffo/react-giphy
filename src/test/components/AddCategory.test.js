import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import AddCategory from "../../components/AddCategory";

const setCategories = jest.fn();
let wrapper = shallow(<AddCategory setCategories={setCategories} />);

beforeEach(() => {
  jest.clearAllMocks();
  wrapper = shallow(<AddCategory setCategories={setCategories} />);
});

describe("prueba de <AddCategory />", () => {
  test("mostrar el componente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe cambiar el texto", () => {
    const input = wrapper.find("input");
    const value = "Hola mundo";
    input.simulate("change", {
      target: {
        value: value
      }
    });
  });

  test("no debe de postear la informaciÃ³n con submit", () => {
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(setCategories).not.toHaveBeenCalled();
  });

  test("debe llamare el setCategories y limpiar la caja de texto", () => {
    const value = "Hola mundo";

    // simular el inputChange
    wrapper.find("input").simulate("change", {
      target: {
        value: value
      }
    });

    // simular el submit
    wrapper.find("form").simulate("submit", { preventDefault() {} });

    // setCategories se debe haber llamado
    expect(setCategories).toHaveBeenCalled();

    // el valor del input debe estar en ''
    expect( wrapper.find('input').prop('value')).toBe('')
  });
});
