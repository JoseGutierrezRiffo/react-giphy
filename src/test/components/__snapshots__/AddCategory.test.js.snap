// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueba de <AddCategory /> mostrar el componente 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
